tools/lkl/lib/virtio_net_linux_fdnet.c

```diff
@@ -47,9 +47,9 @@ static int linux_fdnet_net_tx(struct lkl_netdev *nd,
 
 	do {
 		ret = writev(nd_fdnet->fd, (struct iovec *)iov, cnt);
-	} while (ret == -1 && errno == EINTR);
+	} while (ret == -1 && (errno == EINTR || errno == EAGAIN));
```

Comment:
It doesn't matter who calls net_tx. By definition, epoll_wait(EPOLLOUT) should unblock when fd is writable again after EAGAIN. Do you see failure caused by EAGAIN? You test hangs?